soc: arm: atmel_sam0: Add missing FPU selection

stephanosio opened the pull request:
The SAM D51, E51, E53 and E54 series SoCs include a single-precision FPU; therefore, the
`CPU_HAS_FPU` symbol should be selected.

stephanosio committed:
soc: arm: samd51: Add missing FPU selection

The SAM D51 series SoCs include a single-precision FPU; therefore, the
`CPU_HAS_FPU` symbol should be selected.

Signed-off-by: Stephanos Ioannidis <root@stephanos.io>

stephanosio committed:
soc: arm: same51: Add missing FPU selection

The SAM E51 series SoCs include a single-precision FPU; therefore, the
`CPU_HAS_FPU` symbol should be selected.

Signed-off-by: Stephanos Ioannidis <root@stephanos.io>

stephanosio committed:
soc: arm: same53: Add missing FPU selection

The SAM E53 series SoCs include a single-precision FPU; therefore, the
`CPU_HAS_FPU` symbol should be selected.

Signed-off-by: Stephanos Ioannidis <root@stephanos.io>

stephanosio committed:
soc: arm: same54: Add missing FPU selection

The SAM E54 series SoCs include a single-precision FPU; therefore, the
`CPU_HAS_FPU` symbol should be selected.

Signed-off-by: Stephanos Ioannidis <root@stephanos.io>

stephanosio committed:
soc: arm: atmel_sam0: Add missing arch selection

The SAM D51, E51 and E53 SoC series Kconfig files were missing the
`ARM` architecture symbol selection.

This symbol must be selected at the SoC level; otherwise, build will
fail because the symbol is not selected by anything and other ARM-
specific configurations depend on it.

Signed-off-by: Stephanos Ioannidis <root@stephanos.io>

ioannisg reviewed (commented):
CPU_HAS_FPU if all fine; I am wondering, now, do we really need to select ARM in the SoC/SoC series?
Could we just do it in CPU_CORTEX_M4, or even, CPU_CORTEX?

stephanosio commented:
> CPU_HAS_FPU if all fine; I am wondering, now, do we really need to select ARM in the SoC/SoC series?
> Could we just do it in CPU_CORTEX_M4, or even, CPU_CORTEX?

@ioannisg For the time being, we do; it turns out `CPU_CORTEX_M4` (or `CPU_CORTEX` for that matter) does not select `ARM`.

All the other SoCs `select ARM` too.

ioannisg commented:
> > CPU_HAS_FPU if all fine; I am wondering, now, do we really need to select ARM in the SoC/SoC series?
> > Could we just do it in CPU_CORTEX_M4, or even, CPU_CORTEX?
> 
> @ioannisg For the time being, we do; it turns out `CPU_CORTEX_M4` (or `CPU_CORTEX` for that matter) does not select `ARM`.
> 
> All the other SoCs `select ARM` too.

Yes, I notice that. Do you think we should be able to select ARM in CPU_CORTEX and clean up SoC definitions?

stephanosio commented:
> Yes, I notice that. Do you think we should be able to select ARM in CPU_CORTEX and clean up SoC definitions?

I am not necessarily against it; but, other arch SoCs seem to explicitly select the arch symbol as well (e.g. x86 SoCs select `X86`, RISC-V SoCs select `RISCV`, and so on), so I think it would be better to leave `select ARM` for now.

ioannisg commented:
>I am not necessarily against it; but, other arch SoCs seem to explicitly select the arch symbol as well (e.g. x86 SoCs select X86, RISC-V SoCs select RISCV, and so on), so I think it would be better to leave select ARM for now.

Let's leave this for now, since it is outside the scope of the PR. But I think that since in ARM we do have a symbol hierarchy, we should honor and use it:
ARM SoC: select CORTEX_M_variant -> select CORTEX_M -> select ARM.

I am not sure if the other ARCHes have such an hierarchy, to make use of it.

ioannisg commented:
Could you please, elaborate a bit more on the commit message:
>This symbol must be selected; otherwise, configuration will fail.

stephanosio commented:
> Could you please, elaborate a bit more on the commit message:
> 
> > This symbol must be selected; otherwise, configuration will fail.

Done

stephanosio committed:
dts: arm: atmel: samd5x: Fix CPU compatible

The SAM D5x and E5x series SoCs include the "Cortex-M4F" core, not the
FPU-less "Cortex-M4" core.

This commit fixes the incorrect CPU compatible for the SAM D5x and E5x
SoCs (note that this file is included by `same5x.dtsi`, which is used
as the base device tree header for the SAM E5x series).

Signed-off-by: Stephanos Ioannidis <root@stephanos.io>

ioannisg commented:
> > Could you please, elaborate a bit more on the commit message:
> > > This symbol must be selected; otherwise, configuration will fail.
> 
> Done

These SoCs do not have in-tree boards, is that right @stephanosio ?

stephanosio commented:
> > > Could you please, elaborate a bit more on the commit message:
> > > > This symbol must be selected; otherwise, configuration will fail.
> > 
> > 
> > Done
> 
> These SoCs do not have in-tree boards, is that right @stephanosio ?

@ioannisg Correct. That is also why that went unnoticed.

stephanosio commented:
Added a commit to fix the FPU-related dts spec.
